internal/http/server.go

```diff
@@ -50,18 +59,38 @@ func (s *TechServer) Start(ctx context.Context) error {
 }
 
 // Close shutdowns the HTTP server
-func (s *TechServer) Close(ctx context.Context) error {
+func (s *Server) Close(ctx context.Context) error {
 	return s.httpServer.Shutdown(ctx)
 }
 
-func getTechHttpHandler(pprofEnabled bool, logger logger.LoggerInterface) http.Handler {
+func getHttpHandler(
+	debugEnabled bool,
+	pprofEnabled bool,
+	logger logger.LoggerInterface,
+	debugger Debugger,
+) http.Handler {
 	livenessHandler := handler.NewLiveness(logger)
 
 	router := mux.NewRouter()
 	router.Methods(http.MethodGet).Path("/metrics").Handler(promhttp.Handler())
 	router.Methods(http.MethodGet).Path("/liveness").Handler(livenessHandler)
 	router.Methods(http.MethodGet).Path("/readiness").Handler(livenessHandler)
 
+	if debugEnabled {
```

Comment:
If think it is unnecessary to have a boolean here. Just check if the debugger instance is not nil.